java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/Platform.java

```diff
@@ -0,0 +1,160 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.jni.connector;
+
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+/**
+ * Reference to Apache Spark with some customization
+ */
+public final class Platform {
+
+    private static final Unsafe _UNSAFE;
+
+    public static final int BOOLEAN_ARRAY_OFFSET;
+
+    public static final int BYTE_ARRAY_OFFSET;
+
+    public static final int SHORT_ARRAY_OFFSET;
+
+    public static final int INT_ARRAY_OFFSET;
+
+    public static final int LONG_ARRAY_OFFSET;
+
+    public static final int FLOAT_ARRAY_OFFSET;
+
+    public static final int DOUBLE_ARRAY_OFFSET;
+
+    public static int getInt(Object object, long offset) {
+        return _UNSAFE.getInt(object, offset);
+    }
+
+    public static void putInt(Object object, long offset, int value) {
+        _UNSAFE.putInt(object, offset, value);
+    }
+
+    public static boolean getBoolean(Object object, long offset) {
+        return _UNSAFE.getBoolean(object, offset);
+    }
+
+    public static void putBoolean(Object object, long offset, boolean value) {
+        _UNSAFE.putBoolean(object, offset, value);
+    }
+
+    public static byte getByte(Object object, long offset) {
+        return _UNSAFE.getByte(object, offset);
+    }
+
+    public static void putByte(Object object, long offset, byte value) {
+        _UNSAFE.putByte(object, offset, value);
+    }
+
+    public static short getShort(Object object, long offset) {
+        return _UNSAFE.getShort(object, offset);
+    }
+
+    public static void putShort(Object object, long offset, short value) {
+        _UNSAFE.putShort(object, offset, value);
+    }
+
+    public static long getLong(Object object, long offset) {
+        return _UNSAFE.getLong(object, offset);
+    }
+
+    public static void putLong(Object object, long offset, long value) {
+        _UNSAFE.putLong(object, offset, value);
+    }
+
+    public static float getFloat(Object object, long offset) {
+        return _UNSAFE.getFloat(object, offset);
+    }
+
+    public static void putFloat(Object object, long offset, float value) {
+        _UNSAFE.putFloat(object, offset, value);
+    }
+
+    public static double getDouble(Object object, long offset) {
+        return _UNSAFE.getDouble(object, offset);
+    }
+
+    public static void putDouble(Object object, long offset, double value) {
+        _UNSAFE.putDouble(object, offset, value);
+    }
+
+    public static void freeMemory(long address) {
+        _UNSAFE.freeMemory(address);
+    }
+
+    public static long reallocateMemory(long address, long oldSize, long newSize) {
+        long newMemory = _UNSAFE.allocateMemory(newSize);
```

Comment:
what's the memory allocate from ? via malloc or mmap ?

Reply:
malloc

Reply:
Is it calls our hooked malloc ?

Reply:
it's a java native native method

Reply:
We need to be clear that the memory we are alloc is in full control.

JVM memory we can control through JVM parameters. 
Large blocks of memory requests need to be recorded by the memory tracker.

If unsafe.allocate does not request memory via our malloc (eg: ptmalloc/vmalloc), 
we need to implement an additional memory allocation function

Reply:
done